var/spack/repos/builtin/packages/dihydrogen/package.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Dihydrogen(CMakePackage, CudaPackage):
+    """DiHydrogen is the second version of the Hydrogen fork of the
+       well-known distributed linear algebra library,
+       Elemental. DiHydrogen aims to be a basic distributed
+       multilinear algebra interface with a particular emphasis on the
+       needs of the distributed machine learning effort, LBANN."""
+
+    homepage = "https://github.com/LLNL/DiHydrogen.git"
+    git      = "https://github.com/LLNL/DiHydrogen.git"
+
+    maintainers = ['bvanessen']
+
+    version('master', branch='master')
+
+    variant('al', default=True,
+            description='Builds with Aluminum communication library')
+    variant('developer', default=False,
+            description='Enable extra warnings and force tests to be enabled.')
+    variant('half', default=False,
+            description='Enable FP16 support on the CPU.')
+    variant('legacy', default=False,
+            description='Enable the legacy DistConv code branch.')
+    variant('nvshmem', default=False,
+            description='Builds with support for NVSHMEM')
+    variant('openmp', default=False,
+            description='Enable CPU acceleration with OpenMP threads.')
+    variant('rocm', default=False,
+            description='Enable ROCm/HIP language features.')
+    variant('shared', default=True,
+            description='Enables the build of shared libraries')
+
+    # Override the default set of CUDA architectures with the relevant
+    # subset from lib/spack/spack/build_systems/cuda.py
+    cuda_arch_values = [
+        '60', '61', '62',
+        '70', '72', '75',
+    ]
+    variant('cuda_arch',
+            description='CUDA architecture',
+            values=spack.variant.auto_or_any_combination_of(*cuda_arch_values))
+
+    depends_on('mpi')
+    depends_on('catch2', type='test')
+
+    depends_on('aluminum', when='+al ~cuda')
+    depends_on('aluminum +gpu +nccl +mpi_cuda', when='+al +cuda')
+
+    depends_on('cuda', when=('+cuda' or '+legacy'))
+    depends_on('cudnn', when=('+cuda' or '+legacy'))
+    depends_on('cub', when=('+cuda' or '+legacy'))
+
+    # Legacy builds require cuda
+    conflicts('~cuda', when='+legacy')
+
+    depends_on('half', when='+half')
+
+    generator = 'Ninja'
+    depends_on('ninja', type='build')
+    depends_on('cmake@3.14.0:', type='build')
+
+    illegal_cuda_arch_values = [
+        '10', '11', '12', '13',
+        '20', '21',
+        '30', '32', '35', '37',
+        '50', '52', '53',
+    ]
+    for value in illegal_cuda_arch_values:
+        conflicts('cuda_arch=' + value)
```

Comment:
Since you override the cuda_arch variant above, I think this is already handled. You can test it though.

Reply:
It is handled, and so the code is redundant.  I mostly wanted to put it in as a double check / reminder for when the whitelist is updated.

Reply:
@adamjstewart is this okay to keep or do I have to take it out?

Reply:
You're the maintainer, so you get to decide 😃